var/spack/repos/builtin/packages/py-poetry/package.py

```diff
@@ -42,6 +58,14 @@ class PyPoetry(PythonPackage):
     depends_on("py-keyring@18.0.1:18", when="^python@2.7", type=("build", "run"))
     depends_on("py-keyring@20.0.1:20", when="^python@3.5", type=("build", "run"))
     depends_on("py-keyring@21.2.0:21", when="@1.1.12 ^python@3.6:3", type=("build", "run"))
-    depends_on("py-keyring@21.2.0:", when="@1.1.13 ^python@3.6:3", type=("build", "run"))
+    depends_on("py-keyring@21.2.0:", when="@1.1.13: ^python@3.6:3", type=("build", "run"))
```

Comment:
Should be no condition on Python version, at least in the latest version. Also, note that Spack merges all these dependency constraints, not overwrites them. So right now, you're saying that poetry depends on keyring@20 AND keryring@21: which doesn't make a lot of sense